### PR TITLE
UHF-8216: News RSS feed filters

### PR DESCRIPTION
**Pull request by xkhaven**
# [UHF-8216](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8216)

News RSS feed for news archive using filters

## What was done

* Added link to RSS feed based on selected filters
* Replaced custom datatype for main image with a custom processor
* Reworked aggregations into `multi_term` queries
* Unified query logic between RSS feed and react app

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-8216_News_RSS_feed_filters`
  * `make fresh`
* Run `make drush-cr`

Set up and rebuild Elastic index:
- Run `make up`. Check which port your elastic is running in with `make ps`
- Add this ☝️ to your .env file like so: `ELASTIC_PROXY_URL=http://localhost:[ELASTIC_PORT]`
- Run `make up` again to apply

Create / rebuild index:
- Run `drush sapi-rt; drush sapi-c; drush sapi-i` 

## How to test

Navigate to https://helfi-etusivu.docker.so/fi/uutiset to view the app and test it.

React App:
* [x] Open news archive and add / change some filters
* [x] RSS Feed subscribe link updates with selected / changed filters
* [x] RSS Feed url updates on `<head>` section while changing the filters
* [x] RSS Feed subscribe link opens correct path for current language
* [x] Check that code follows our standards

RSS feed:
Easiest way to test this is to use Postman or similar tool that lets you see the XML from RSS feed. Both the RSS feed and archive app should yield the same results with filters applied. Test this by making some searches with the archive and then copypasting the query string to the RSS url https://helfi-etusivu.docker.so/fi/uutiset/rss. For example, `https://helfi-etusivu.docker.so/fi/uutiset/rss?topic[0]=397&groups[0]=2` and `https://helfi-etusivu.docker.so/fi/uutiset?topic[0]=397&groups[0]=2` should give you the same set of results.

## Designers review

* [X] This PR does not need designers review

## Other PRs

* [UHF-8216: Styles for News RSS feed link](https://github.com/City-of-Helsinki/drupal-hdbt/pull/645)

[UHF-8216]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Jussiles — COMMENTED** [review]
I pushed one change for views and now the RSS feed seems to work for me. Test it if you can verify that it works now. Otherwise everything seems to work. 🦜

File: conf/cmi/search_api.index.news.yml [hunk 1]
```diff
@@ -109,6 +104,48 @@ field_settings:
     dependencies:
       config:
         - field.storage.node.field_short_title
+  main_image_url:
+    label: Pääkuva
```
Comment (rpnykanen) [inline conf/cmi/search_api.index.news.yml:108]: label language

File: conf/cmi/search_api.index.news.yml [hunk 2]
```diff
@@ -109,6 +104,48 @@ field_settings:
...
+    property_path: main_image_url
+    type: string
+  name:
+    label: Nimi
```
Comment (rpnykanen) [inline conf/cmi/search_api.index.news.yml:112]: label language

File: conf/cmi/search_api.index.news.yml [hunk 3]
```diff
@@ -109,6 +104,48 @@ field_settings:
...
+    datasource_id: 'entity:node'
+    property_path: 'field_news_item_tags:entity:tid'
+    type: integer
+    dependencies:
+      config:
+        - field.storage.node.field_news_item_tags
+      module:
+        - taxonomy
   published_at:
     label: Julkaisuaika
```
Comment (rpnykanen) [inline conf/cmi/search_api.index.news.yml:150]: language

File: conf/cmi/search_api.index.news.yml [hunk 4]
```diff
@@ -117,6 +154,14 @@ field_settings:
     dependencies:
       module:
         - publication_date
+  tid:
+    label: 'Termin ID'
```
Comment (rpnykanen) [inline conf/cmi/search_api.index.news.yml:158]: language

File: conf/cmi/search_api.index.news.yml [hunk 5]
```diff
@@ -145,6 +190,14 @@ field_settings:
     dependencies:
       module:
         - node
+  vid:
+    label: Luokittelu
```
Comment (rpnykanen) [inline conf/cmi/search_api.index.news.yml:194]: Language

File: conf/cmi/views.view.react_news_feed.yml [hunk 6]
```diff
@@ -22,6 +24,7 @@ display:
     display_plugin: default
     position: 0
     display_options:
+      title: Uutiset
```
Comment (rpnykanen) [inline conf/cmi/views.view.react_news_feed.yml:27]: language

File: conf/cmi/views.view.react_news_feed.yml [hunk 7]
```diff
@@ -1826,23 +1755,31 @@ display:
...
+                pubDate: published_at
+            content:
+              views_rss_core:
+                encoded: ''
       defaults:
         fields: false
         sorts: false
       display_extenders: {  }
-      path: news/react/rss
+      path: news/subscribe/rss
```
Comment (rpnykanen) [inline conf/cmi/views.view.react_news_feed.yml:1781]: is this url relevant ?

**Review by rpnykanen — APPROVED** [review]

**Pull request merged**